src/interpolate/Interpolate_transform.re

```diff
@@ -0,0 +1,76 @@
+type cssTransform = {
+  transform: string,
+  transformProperty: Js.Nullable.t(string),
+};
+
+[@bs.deriving jsConverter]
+type transformProperties = [
+  | `translate
+  | `translateX
+  | `translateY
+  | `translateZ
+  | `skew
+  | `skewX
+  | `skewY
+  | `skewZ
+  | `rotate
+  | `rotateX
+  | `rotateY
+  | `rotateZ
+  | `scale
+  | `scaleX
+  | `scaleY
+  | `scaleZ
+  | `perspective
+];
+
+let parseTransform = transform => {
+  switch (String.index(transform, '(')) {
+  | idx =>
+    /* Decompose the translate string into the translate property, i.e. translateX, skew, etc.
+       Then, extract the value, i.e. 20px, 50deg, so we can begin to interpolate it. */
+    let property =
+      Js.String.substring(~from=0, ~to_=idx, transform)
+      ->transformPropertiesFromJs;
+    let value =
+      Js.String.substring(
+        ~from=idx + 1,
+        ~to_=String.index_from(transform, idx + 1, ')'),
```

Comment:
So here we're basically hedging a bet that a user is giving us a string that has both a closing and opening parenthesis. We _could_ be even more type safe here, but it feels a bit overkill right now. Hopefully people aren't thinking that this lib should handle `transform: translateX(4px` (no closing parenthesis).